src/components/visualizations/chart/legend/ColorLegend.tsx

```diff
@@ -0,0 +1,90 @@
+// (C) 2007-2020 GoodData Corporation
+import * as React from "react";
+import * as classNames from "classnames";
+import {
+    IColorLegendBox,
+    IColorLegendConfig,
+    IHeatmapLegendLabel as IColorLegendLabel,
+    getColorLegendConfiguration,
+} from "./helpers";
+import { TOP } from "./PositionTypes";
+import { IColorLegendItem } from "../../typings/legend";
+import {
+    IntlTranslationsProvider,
+    ITranslationsComponentProps,
+} from "../../../core/base/TranslationsProvider";
+
+export interface IColorLegendProps {
+    colorData: IColorLegendItem[];
+    isSmall: boolean;
+    format?: string;
+    numericSymbols: string[];
+    position: string;
+}
+
+function ColorLabels(labels: IColorLegendLabel[]): JSX.Element {
+    return (
+        <div className="labels">
+            {labels.map((item: IColorLegendLabel) => {
+                const { key, label, style } = item;
+                return (
+                    <span key={key} style={style}>
+                        {label}
+                    </span>
+                );
+            })}
+        </div>
+    );
+}
+
+function ColorBoxes(boxes: IColorLegendBox[]): JSX.Element {
+    return (
+        <div className="boxes">
+            {boxes.map((box: IColorLegendBox) => {
+                const classes = classNames("box", box.class);
+                const { key, style } = box;
+                return <span className={classes} key={key} style={style} />;
+            })}
+        </div>
+    );
+}
+
+export function ColorLegend(colorLegendProps: IColorLegendProps): JSX.Element {
+    const { colorData, format, numericSymbols, isSmall, position } = colorLegendProps;
+
+    const config: IColorLegendConfig = getColorLegendConfiguration(
+        colorData,
+        format,
+        numericSymbols,
+        isSmall,
+        position,
+    );
+    const classes = classNames(...config.classes);
+    const renderLabelsFirst = config.position === TOP;
+    const { boxes, labels } = config;
+
+    return (
+        <div className={classes}>
+            {renderLabelsFirst && ColorLabels(labels)}
+            {ColorBoxes(boxes)}
+            {!renderLabelsFirst && ColorLabels(labels)}
+        </div>
+    );
+}
+
+export function IntlColorLegend(colorLegendProps: Partial<IColorLegendProps>): JSX.Element {
+    const { colorData, format } = colorLegendProps;
+    return (
+        <IntlTranslationsProvider>
+            {(props: ITranslationsComponentProps) => (
```

Comment:
it could be new hook like
```
export function useTranslationProvider() {
    const intl = useIntl();
    return {
        numericSymbols: getNumericSymbols(intl),
        emptyHeaderString: getEmptyHeaderString(intl),
    };
}
```